examples/deviceid.rs

```diff
@@ -0,0 +1,55 @@
+//! Transmit the device ID repeatedly via USB serial.
+
+#![no_std]
+
+extern crate numtoa;
+extern crate particle_hal as hal;
+#[macro_use]
+extern crate photon;
+
+app! {
+    setup: setup,
+    loop: loop_,
+}
+
+use hal::delay;
+use hal::usb;
+use hal::deviceid;
+use numtoa::NumToA;
+
+const PERIOD: u32 = 1000; // ms
+const BAUD_RATE: u32 = 115_200;
+
+static mut DEVICE_ID: [u8; 12] = [0; 12];
```

Comment:
These variables could be allocated on the `loop_`'s stack. `let mut device_id: [0; 12]`.